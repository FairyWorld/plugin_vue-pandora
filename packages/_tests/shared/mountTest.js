import { mount } from '@vue/test-utils';
export default function mountTest(component) {
  describe(`mount and unmount`, () => {
    // https://github.com/ant-design/ant-design/pull/18441
    it(`component could be updated and unmounted without errors`, () => {
      const wrapper = mount(component,
        { 
          sync: false, 
          attachTo: 'body' 
        },
      );
      expect(() => {
        wrapper.vm.$forceUpdate();
        wrapper.unmount();
      }).not.toThrow();
    });
  });
}
