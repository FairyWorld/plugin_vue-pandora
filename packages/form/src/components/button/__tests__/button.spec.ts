import { mount } from '@vue/test-utils'
import mountTest from '@/packages/_tests/shared/mountTest'
import Button from '@/packages/form/src/components/button/index'

describe('Button.vue', () => {
  mountTest(Button);     
  it('renders correctly', () => {
    const wrapper = mount(Button);
    expect(wrapper.html()).toMatchSnapshot();
  });
})
