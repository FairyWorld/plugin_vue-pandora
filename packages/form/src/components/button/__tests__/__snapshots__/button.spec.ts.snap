// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button.vue renders correctly 1`] = `
<button label="default Btn" class="ant-btn ant-btn-default" type="button">
  <!----><span>default Btn</span>
</button>
`;
